From 0b5e5bd8b6400f68817e1255e04c19023efd6b95 Mon Sep 17 00:00:00 2001
From: MSe1969 <mse1969@posteo.de>
Date: Sun, 12 Aug 2018 13:27:29 +0200
Subject: [PATCH] Different sign method in build script [2/2]

The original method, as documented by Google / LineageOS, sometimes causes
errors when executing the 'ota_from_target_files' python script.

Therefore, the combination of PRODUCT_DEFAULT_DEV_CERTIFICATE in common.mk
of vendor/lineage and setting a symlink testkey.* => releasekey.* in the
certs directory to allow selinux to build has been chosen, similar to
the docker image in github.com/lineageos4microg.

Change-Id: I4b8b26ef3f6d54f7d4ee1c570c1737779b6b8884
---
 config/common.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index a30fcf68..6cd9ea4a 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -1,5 +1,12 @@
 PRODUCT_BRAND ?= LineageOS

+# Sign with own key
+ifneq ($(OWN_KEYS_DIR),)
+PRODUCT_DEFAULT_DEV_CERTIFICATE := $(OWN_KEYS_DIR)/releasekey
+PRODUCT_OTA_PUBLIC_KEYS := $(OWN_KEYS_DIR)/releasekey
+PRODUCT_EXTRA_RECOVERY_KEYS := $(OWN_KEYS_DIR)/releasekey
+endif
+
 PRODUCT_BUILD_PROP_OVERRIDES += BUILD_UTC_DATE=0

 ifeq ($(PRODUCT_GMS_CLIENTID_BASE),)
--
2.11.0

